### DISCUSSION TITLE
Raising the limit of ST files or making it arbitrary

### DISCUSSION THREAD

**Discussion by Jay-xs** [ikemen-engine/Ikemen-GO#1704] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

Sometimes is hard to locate what want and keep track of things in a very big code. Specially if you like to separate each type of state in a different file.

### Describe the solution you'd like

Making the number of ST files higher or arbitrary.  I know 11 st files is already a lot, but more would be great! It'd make things way easier to keep track of.

### Additional context

If you're writing a character that has two different Modes, eg: Ken/Violent Ken, and like... violent ken has a full set of "supers". Keeping them in the same state as your normal mode to save space, makes things very hard to follow. Specially if you states doesn't follow a linear number. Not to mention if you like to separate your AI into different files such as  AI hard, AI medium, AI easy etc... As for myself, i like to keep my "Effect Helpers" in a separate file too, so i can organize them in a much better way. Characters tend to have a lot of Special Effects... And if you make a Helper state for each one of them, it quickly becomes absurdly big. So i tend to separate them into files such as: Fight Effects.st, System Effects.st, Camera Control.st, Animations .st, cutscenes.st, special intros.st etc...
I just happens that 11 ST files feels like it's not a very big number to me... Also some IDEs(Mainly FF)'s code editor tend to break, if you have a very looong code file.
keeping things like that in the same ZSS/ST, to save space, makes it very hard to navigate, and organize. So if you can have an arbritrary number of ST files, you can keep all of your states in a specific category. You can have a very long code for your Hard AI, Medium AI etc... as well as having a separate ST for supers with Animations.

### Search terms

st_files, st, files, st1, st0, limit

**Comment by leandrooffline** [comment]
In old versions was possible st10, st11 etc.
Or does removed or its a bug.
(I want more st's too)

**Comment by Kiwi1444** [comment]
Problems start at st10. It doesn't matter how many st files you have in total. The system can't read states from st10 and everything higher in number. Today I tried st10 and ran into problems.

**Comment by shinra358** [comment]
yes pls.

**Comment by K4thos** [comment, marked as answer]
done in #2264
